rabbitmq/tests/test_unit.py

```diff
@@ -147,3 +147,25 @@ def test_config(check, test_case, extra_config, expected_http_kwargs):
         http_wargs.update(expected_http_kwargs)
 
         r.get.assert_called_with('http://localhost:15672/api/connections', **http_wargs)
+
+
+@pytest.mark.unit
+def test_nodes_unit(aggregator, check):
+
+    # default, node metrics are collected
+    check.check(common.CONFIG)
+
+    for m in metrics.COMMON_METRICS:
+        aggregator.assert_metric(m, count=1)
+
+    aggregator.reset()
+
+    # node metrics collection disabled in config, node metrics should not appear
+    check.check(common.CONFIG_NO_NODES)
```

Comment:
split into separate tests. The check method should not rely on the passed in `instance` because this parameter is deprecated. The instance should get fixed at constructor time. Older integrations may not be following this but we should not make the problem worse.

Reply:
Sounds good!